x/mint/keeper/keeper_test.go

```diff
@@ -719,3 +719,348 @@ func (suite *KeeperTestSuite) TestDistributeDeveloperRewards() {
+		amountDistributed sdk.Int
+
+		// Expected outputs
+		expectedDistributedAmount sdk.Int
+		expectedPersistedInStore  sdk.Dec
+		expectErr                 bool
+	}{
+		"valid case: 100.6 - 100 + 0.2 = 0.8; none-distrubuted and persisted until next epoch": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			preExistingDelta:  sdk.NewDecWithPrec(2, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.ZeroInt(),
+			expectedPersistedInStore:  sdk.NewDecWithPrec(8, 1),
+		},
+		"valid case: 100.6 - 100 + 0.4 = 1; all distributed and none persisted": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			preExistingDelta:  sdk.NewDecWithPrec(4, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.OneInt(),
+			expectedPersistedInStore:  sdk.ZeroDec(),
+		},
+		"valid case: 100.8 - 100 + 0.4 = 1.2; 1 distributed, 0.2 persisted in store; mint module account": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(4, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1008, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.OneInt(),
+			expectedPersistedInStore:  sdk.NewDecWithPrec(2, 1),
+		},
+		"valid case: 100 - 100 + 0 = 0; none distributed or persisted in store; mint module account": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.ZeroDec(),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDec(100)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectedDistributedAmount: sdk.ZeroInt(),
+			expectedPersistedInStore:  sdk.ZeroDec(),
+		},
+		"invalid module account name - error": {
+			moduleAccountName: poolincentivestypes.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(2, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectErr: true,
+		},
+		"provisions are less than amount distributed - error": {
+			moduleAccountName: types.ModuleName,
+			preExistingDelta:  sdk.NewDecWithPrec(2, 1),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(101),
+
+			expectErr: true,
+		},
+		"attempted to distribute more than developer vesting module account balance - error": {
+			moduleAccountName: types.DeveloperVestingModuleAcctName,
+			preExistingDelta:  sdk.NewDecFromInt(sdk.NewInt(keeper.DeveloperVestingAmount)).Add(sdk.OneDec()),
+			provisions:        sdk.NewDecCoinFromDec(sdk.DefaultBondDenom, sdk.NewDecWithPrec(1006, 1)),
+			amountDistributed: sdk.NewInt(100),
+
+			expectErr: true,
+		},
+	}
+	for name, tc := range tests {
```

Comment:
Same as above, I think we need a test case where we test the accumulation of the truncation delta over multiple epochs to ensure it works as expected (0.1 > 0.6 > 1.1 or something like that)

Reply:
I don't understand the value of running this several times in a row in a unit test because, instead, we can make sure that a single run unit test covers all combinations of input parameters and relevant store values.

There are functional and integrations test things like this. We do have a functional test that helps to ensure that multi-epoch values are correct:
https://github.com/osmosis-labs/osmosis/blob/d7caed8acfe3b756134c42c839bfca1d6225cb1b/x/mint/keeper/hooks_test.go#L444

It's in the original large PR still